feat(lorem): add character length parameter to lorem.text

inkedtree opened the pull request:
Adds a character length parameter to `faker.lorem.text` as described in #1546.

Changes made:

- ~~Added a new parameter `length?: number | { min: number, max: number }` such that `lorem.text` returns text that has a particular length if the parameter is passed~~
- Added a new optional options object `{length: number | { min: number, max: number }}` such that `lorem.text` returns text that has a particular length if the parameter is passed
- Added a new optional quick primitive argument such that `lorem.text` returns text that the length passed as argument
- Added random seeded tests for the new parameter to `lorem.spec.ts` and updated its snapshot

Closes #1546

Commit:
add character length parameter to lorem.text

ST-DDT commented on test/__snapshots__/lorem.spec.ts.snap:
```diff
@@ -97,9 +97,9 @@ exports[`lorem > 42 > slug > with length range 1`] = `"minus-a-qui-tempore-quo-p
 exports[`lorem > 42 > text > noArgs 1`] = `"Qui tempore quo perspiciatis perspiciatis quia in quia architecto ipsa. Pariatur veniam unde fugit facilis mollitia. Ipsa reiciendis nam. Earum dolorem alias qui asperiores qui natus aliquam iste. Consequatur eum sit dolore rem iste aliquid."`;
 
-exports[`lorem > 42 > text > with length 1`] = `"Qui tempore quo perspiciatis perspiciatis quia in quia architecto ipsa. Pariatur veniam unde fugit facilis mollitia. Ipsa reiciendis nam. Earum dolorem alias qui asperiores qui natus aliquam iste. Consequatur eum sit dolore rem iste aliquid."`;
+exports[`lorem > 42 > text > with length 1`] = `" Qui temp."`;
```
IMO the text should not start with a whitespace.

inkedtree replied:
Thank you. I fixed this.

matthewmayer commented on src/modules/lorem/index.ts:
```diff
@@ -318,6 +318,10 @@ export class LoremModule {
   /**
    * Generates a random text based on a random lorem method.
    *
+   * @param length The length (range) of text to generate.
```
Not sure we should introduce new single arguments like this, we are trending towards putting everything in options objects. What if we need another parameter later?

inkedtree replied:
I removed the single argument and added an option object.

xDivisionByZerox replied:
While we should have an options object for future extendability, we often allow the user to provide a "quick primitive argument" for the option that is most likely to be used.

Examples:
```ts
// equivalent
faker.finance.accountNumber(5);
faker.finance.accountNumber({ length: 5 });

// equivalent
faker.location.zipCode('###');
faker.location.zipCode({ format: '###' }});

// equivalent
faker.number.int(1337);
faker.number.int({ max: 1337 }});

// equivalent
faker.string.alphanumeric(1337);
faker.string.alphanumeric({ length: 1337 }});

// equivalent
faker.word.adverb(42);
faker.word.adverb({ length: 42 }});
```

inkedtree replied:
Thank you for the examples. I added the quick primitive argument again.

matthewmayer commented on test/lorem.spec.ts:
```diff
@@ -325,6 +325,85 @@ describe('lorem', () => {
+        it('should return null for a negative length', () => {
+          const actual = faker.lorem.text(-3);
+
+          expect(actual).toBeNull();
```
I feel this should throw as is probably programmer error.

ST-DDT replied:
IIRC we return empty strings for negative size strings as well.

inkedtree replied:
I am not sure which behaviour is better. In faker.string.alpha an empty string is returned, if I understand the code correctly. But I do not find this intuitive. Currently the method throws an error in this case. Is this fine?

xDivisionByZerox replied:
Regardless of the decision, this should be discussed in a separate issue.

matthewmayer commented on test/lorem.spec.ts:
```diff
@@ -325,6 +325,85 @@ describe('lorem', () => {
+        });
+
+        it('should return an empty string for [0, 32]', () => {
+          const actual = faker.lorem.text({ min: 0, max: 32 });
```
This doesn't seem that intuitive. I would expect to get string's with random lengths between 0 and 32

inkedtree replied:
Thank you. I fixed this.

matthewmayer commented on test/lorem.spec.ts:
```diff
@@ -325,6 +325,85 @@ describe('lorem', () => {
+
+        it('should return null for [3, 1]', () => {
+          const actual = faker.lorem.text({ min: 3, max: 1 });
+
```
This should also throw

inkedtree replied:
Thank you. I fixed this.

xDivisionByZerox replied:
This should return an empty string to have a consistent behavior across other methods.

inkedtree replied:
Now, it returns an empty string.

matthewmayer replied:
I feel {min:3, max:1} is almost certainly due to programmer error (Ie they accidentally reversed it) so it's better to throw than return an empty string.

xDivisionByZerox replied:
I checked. Methods in the StringModule do indeed throw if `max` is smaller than `min`. Internally, they use `helpers.rangeToNumber`, which uses `number.int` under the hood. `number.int` covers the throwing behavior. That's my bad.

inkedtree replied:
Oh. Now, it throws again :)

Commit:
resolve issues noted in the review of ST-DDT and matthewmayer

Commit:
fix example in comment of lorem.text

Commit:
return an empty string for an ill defined interval instead of throwing an error

Commit:
fix typo

Commit:
add quick primitive argument to lorem.text

matthewmayer commented on src/modules/lorem/index.ts:
```diff
@@ -326,10 +332,32 @@ export class LoremModule {
    * // Dolor tempora iusto.'
+   * faker.lorem.text(14) // 'Doloribus aut.' (14 characters)
+   * faker.lorem.text({length: 14}) // 'Doloribus aut.' (14 characters)
+   * faker.lorem.text({min: 10, max: 15}) // 'autem non quis.' (15 characters)
```
I think {length:x} or {length:{min:x, max:y}} would be clearer. Then you avoid edge cases like what should happen if you pass {length:1, max:2}

inkedtree replied:
Good point. I changed the code.

xDivisionByZerox commented on CONTRIBUTING.md:
```diff
@@ -9,7 +9,7 @@ This is a shorthand for running the following scripts in order:
 - `pnpm install` - installs npm packages defined in package.json
 - `pnpm run generate:locales` - generates locale files
-- `pnpm run generate:api-doc` - generates API documentation
+- `pnpm run generate:api-docs` - generates API documentation
```
While I love that you did spot this mistake, please don't mix concerns of PR. This PR has the concern to implement a feature, not fixing a documentation typo.

ST-DDT replied:
You can create a different PR for that, that way the PRs stay on one topic and are easier to review and merge.

inkedtree replied:
Oh, sorry. I did not pay attention to this. I created a [different PR](https://github.com/faker-js/faker/pull/2264) for that.

Commit:
it throws an error if min is greater than max instead of returning an empty string

Commit:
the length parameter in lorem.text is a number or an object with min and max instead of being an object with all three properties

matthewmayer commented on test/lorem.spec.ts:
```diff
@@ -325,6 +325,125 @@ describe('lorem', () => {
+          expect(actual).toBe('');
+        });
+
+        it('should return an empty string for [0, 1]', () => {
```
The test is right but the description is wrong, it doesn't always return empty string

inkedtree replied:
I corrected the description.

matthewmayer commented on test/lorem.spec.ts:
```diff
@@ -325,6 +325,125 @@ describe('lorem', () => {
+          expect(actual).toHaveLength(77);
+        });
+
+        it('should return an empty string for [3, 1]', () => {
```
Again desc is wrong

inkedtree replied:
I corrected the description.

matthewmayer commented on src/modules/lorem/index.ts:
```diff
@@ -340,7 +371,54 @@ export class LoremModule {
+      );
+    }
+
+    const effectiveMin = Math.max(min, 0);
```
I think it would simplify the implementation to just throw for a negative min too

inkedtree replied:
You're right. I also think the implementation is simpler now.

Commit:
corrected descriptions in lorem.spec.ts and lorem.text throws an error for negative min

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

xDivisionByZerox commented on src/modules/lorem/index.ts:
```diff
@@ -326,10 +332,35 @@ export class LoremModule {
+   * faker.lorem.text({length: 14}) // 'Doloribus aut.' (14 characters)
+   * faker.lorem.text({length: {min: 10, max: 15}}) // 'autem non quis.' (15 characters)
+   * faker.lorem.text({length: {min: 10, max: 12}}) // 'autem non.' (10 characters)
+   * faker.lorem.text({length: {min: 3, max: 10}}) // 'autem.' (6 characters)
```
I dont think we need the explicit character information at the end. We also dont need that many repetitive examples:

```suggestion
   * faker.lorem.text(14) // 'Doloribus aut.'
   * faker.lorem.text({length: 14}) // 'Doloribus aut.'
   * faker.lorem.text({length: {min: 10, max: 15}}) // 'autem non quis.'
```

matthewmayer reviewed (commented):
I think the description of `text()` in the overview could also be modified to reflect this can be used to produce text of a certain length

```The generic [`text()`](https://fakerjs.dev/api/lorem.html#text) method can be used to generate some text between one sentence and multiple paragraphs```

matthewmayer commented:
The behavior of the function when the length parameter is supplied is quite different to the version without.

I'm wondering if therefore it would make sense to just make it a separate method, like:

```
* faker.lorem.characters(14) // 'Doloribus aut.' (14 characters)
* faker.lorem.characters({length: 14}) // 'Doloribus aut.' (14 characters)
* faker.lorem.characters({length: {min: 10, max: 15}}) // 'autem non quis.' (15 characters)
* faker.lorem.characters({length: {min: 10, max: 12}}) // 'autem non.' (10 characters)
```

Commit:
remove explicit character information and repetitive examples from the js-doc for lorem.text

Commit:
Merge branch 'feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text' of github.com:inkedtree/faker into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

Commit:
fix issue after merging and update snapshot for lorem module

ST-DDT commented:
> I'm wondering if therefore it would make sense to just make it a separate method, like:
> 
> ```
> * faker.lorem.characters(14) // 'Doloribus aut.' (14 characters)
> ```

IMO the result aren't characters, but some kind of text/words. Maybe rename the `length` parameter to `characters` instead?

matthewmayer commented:
its not purely a naming issue, the behavior of text() is a little surprising, because depending on which other faker method is called internally sometimes you get sentences separated by newlines, or one big paragraph etc.

<img width="1077" alt="image" src="https://github.com/faker-js/faker/assets/152770/6292fb16-5895-43fd-8762-bd4ae416bbc4">

ST-DDT commented:
Maybe we should deprecate `text` in favor of `sentences` and `paragraphs`. We will discuss this tomorrow, if we have the time.
(and add the characters option to the other methods)

ST-DDT commented:
There seems to be merge conflicts. Could you please fix them?

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

ghost commented:
Sorry for the late reply. I fixed the merge conflicts.

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +324,33 @@ export class LoremModule {
+                /**
+                 * The minimum length of text to generate.
+                 */
+                min?: number;
```
```suggestion
                min: number;
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +324,33 @@ export class LoremModule {
+                /**
+                 * The maximum length of text to generate.
+                 */
+                max?: number;
```
```suggestion
                max: number;
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +361,57 @@ export class LoremModule {
+      return `${this[method]()}`;
+    }
+
+    if (typeof options === 'number') {
+      return this.faker.lorem.text({ length: options });
+    }
```
```suggestion
    if (typeof options === 'number') {
      options = { length: options };
    }
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +324,33 @@ export class LoremModule {
    */
-  text(): string {
+  text(
+    options?:
```
```suggestion
    options:
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +324,33 @@ export class LoremModule {
+                 */
+                max?: number;
+              };
+        }
```
```suggestion
        } = {}
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +361,57 @@ export class LoremModule {
+    if (typeof options === 'number') {
+      return this.faker.lorem.text({ length: options });
+    }
+
```
You can use https://fakerjs.dev/api/helpers.html#rangetonumber to simplify this:

```suggestion

    const length = options.length == null ? null : faker.helpers.rangeToNumber(options.length);

    if (length == null) {
      return `${this[method]()}`;
    }

```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +361,57 @@ export class LoremModule {
     const method = this.faker.helpers.arrayElement(methods);
 
-    return `${this[method]()}`;
+    if (typeof options === 'undefined') {
+      return `${this[method]()}`;
+    }
```
```suggestion
```

Then you don't need this part and the behavior is more predictable.

e.g. `text()` behaves the same as `text({})`

Commit:
feat(lorem): improve code

- apply feedback of code review to make the code more predictable and readable
- if length includes min, it also has to include max, and vice versa
- adapt tests

Commit:
feat(lorem): remove .only and add lorem-snapshot

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

ghost commented:
Thank you. I am not sure if I implemented your feedback correctly, but I tried to do my best.

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -310,6 +311,11 @@ export class LoremModule {
+   * @param options The options for the text to generate or options.length as quick primitive argument.
+   * @param options.length The length (range) of text to generate.
+   * @param options.length.min The minimum length of text to generate.
+   * @param options.length.max The maximum length of text to generate.
```
We dont add `@param` blocks for these doubly nested options.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +361,49 @@ export class LoremModule {
+    if (typeof options === 'number') {
+      options = { length: options };
+    }
+
```
You can use https://fakerjs.dev/api/helpers.html#rangetonumber to simplify the code below:

```suggestion

    if (options.length == null) {
      return this[method]();
    }

    const length = this.faker.helpers.rangeToNumber(options.length);

```

Commit:
feat(lorem): improve jsdoc and simplify code of lorem.text()

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +359,46 @@ export class LoremModule {
+      return `${this[method]()}`;
+    }
+
+    if (typeof options.length === 'number' && options.length < 0) {
+      throw new FakerError(
+        `Length ${options.length} should be a non-negative integer.`
+      );
+    }
+
+    if (typeof options.length !== 'number') {
+      if (options.length.min != null && options.length.min < 0) {
+        throw new FakerError(
+          `Min ${options.length.min} should be a non-negative integer.`
+        );
+      }
+
+      if (options.length.max != null && options.length.max < 0) {
+        throw new FakerError(
+          `Max ${options.length.max} should be a non-negative integer.`
+        );
+      }
+    }
+
```
There is no need to check for these.
1. We don't check for the presence of required values
2. We ignore negative bounds and treat them as zero (for string lengths).

```suggestion
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +359,46 @@ export class LoremModule {
+
+    const length = this.faker.helpers.rangeToNumber(options.length);
+
+    if (length === 0) {
```
```suggestion
    if (length <= 0) {
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +359,46 @@ export class LoremModule {
+      return '';
+    }
+
+    let text = `${this[method]()}`;
+    while (text.length <= length) {
+      text = `${text} ${this[method]()}`;
+    }
```
It is also possible to write this as a do-while-loop, but I'm not sure about that:

```suggestion
    let text = '';
    do {
      text = `${text}${this[method]()} `;
    } while (text.length < length);
```

inkedtree replied:
What is the advantage of doing is this way?

ST-DDT replied:
There are two tiny benefits:

1. The trailing ` ` adds one to the text.length and thus terminates the loop potentially early (this would also work in the normal while loop)
2. The method is invoked in only one location

With the tiny drawback that there is one additional operation with the empty string

ST-DDT commented on test/modules/lorem.spec.ts:
```diff
@@ -325,6 +325,117 @@ describe('lorem', () => {
+          expect(actual).toHaveLength(77);
+        });
+
+        it('should return text with a length in [33, 66]', () => {
+          const actual = faker.lorem.text({ length: { min: 33, max: 66 } });
+
+          expect(actual).toBeTypeOf('string');
+          expect(actual.length).toBeGreaterThanOrEqual(33);
+          expect(actual.length).toBeLessThanOrEqual(66);
+        });
+
+        it('should return text with a length in [0, 51]', () => {
+          const actual = faker.lorem.text({ length: { min: 0, max: 51 } });
+
+          expect(actual).toBeTypeOf('string');
+          expect(actual.length).toBeGreaterThanOrEqual(0);
+          expect(actual.length).toBeLessThanOrEqual(51);
+        });
```
IMO there are a few redundant tests here.

ST-DDT reviewed (commented):
Also, should we stick with `sentence()` (if length is provided) to produce a more consistent text?
(We end with a dot, so it would make sense to have dots in between as well.)

Commit:
feat(lorem): remove unnecessary checks in text()

Commit:
feat(lorem): remove redundant tests

Commit:
feat(lorem): remove tests for checks that were removed

Commit:
feat(lorem): simplify code

ghost commented:
Yes, it would make sense to put dots in between as well.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
+    let text = '';
+    do {
+      text = `${text}${this[method]()} `;
+    } while (text.length <= length + 1);
```
```suggestion
    } while (text.length < length - 1);
```

If `text.length === length - 1` then the substring below doesn't remove anything and appends the '.' string => `text.length === length`. It would end in `foo .` but that would still be the case even if you append another word to it, because it would get removed via the substring call.

---

Should we deduct the `-1` from `length` early?

````ts
// deduct one from the length for the final dot
length--;
````

---

Alternatively see this comment: https://github.com/faker-js/faker/pull/2227#discussion_r1299768500

inkedtree replied:
I implement it similar to the way you described in https://github.com/faker-js/faker/pull/2227#discussion_r1299768500.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
+
+    let text = '';
+    do {
+      text = `${text}${this[method]()} `;
```
If we go for `sentence()` (not sure yet):

```suggestion
      text = `${text}${this.sentence()} `;
```

ST-DDT replied:
Please only use `this.sentence()` here and maybe randomly (e.g. faker.datatype.boolean()) add a newline to simulate paragraphs/lines.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
 
     const method = this.faker.helpers.arrayElement(methods);
```
If we go for `sentence()` (not sure yet):

You can move this to the `if (options.length == null)` block.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
+      text = `${text}${this[method]()} `;
+    } while (text.length <= length + 1);
+
+    return `${text.trim().substring(0, length - 1)}.`;
```
To prevent whitespace followed by a dot:

```suggestion
    return `${text.trim().substring(0, length - 1).replace(/\s$/, '.')}.`;
```

ST-DDT replied:
Alternatively, we could not deduct 1 (here and in the loop) from the length and use `.replace(/.$/, '.')` without appending the final dot.

ST-DDT commented on test/modules/lorem.spec.ts:
```diff
@@ -325,6 +325,77 @@ describe('lorem', () => {
+          expect(actual).toBe('.');
+        });
+
+        it('should return a string of length 0 or length 1 for [0, 1]', () => {
+          const actual = faker.lorem.text({ length: { min: 0, max: 1 } });
+
+          expect(actual).toBeTypeOf('string');
+          expect(actual.length).toBeGreaterThanOrEqual(0);
+          expect(actual.length).toBeLessThanOrEqual(1);
+        });
+
+        it('should return text with length between 0 and 32 for [0, 32]', () => {
+          const actual = faker.lorem.text({ length: { min: 0, max: 32 } });
+
+          expect(actual).toBeTypeOf('string');
+          expect(actual.length).toBeGreaterThanOrEqual(0);
+          expect(actual.length).toBeLessThanOrEqual(32);
+        });
+
+        it('should return a string of length 77 for [77, 77]', () => {
+          const actual = faker.lorem.text({ length: { min: 77, max: 77 } });
+
+          expect(actual).toBeTruthy();
+          expect(actual).toBeTypeOf('string');
+          expect(actual).toHaveLength(77);
+        });
```
IMO these are already covered by `should return text with a length in range [25, 40]'`

ST-DDT commented:
@inkedtree We kind of are unsure what requirements we have for the method/the parameter. And would like to hear your requirements for the method.
This method generates vastly different types of text (some with newlines, some without).
So do we need the length parameter for the specific versions as well? Such as sentences, lines?
Or should we limit the text method to sentences?
Would it get confusing if we have `lines( { length: number})` aka does the length refer to each line or to the entire string?

ghost commented:
@ST-DDT

Sorry, for the late reply. I am not sure, if I understand all your questions correctly.

To be honest, I am not entirely sure what my requirements are after reading about all available methods in the `lorem` module. Originally, I wanted to have a method that generates sentences with a particular length. So, basically a paragraph, I guess. But now I am not sure, if it wouldn't be better to add the parameter to `paragraph(s)` for example in this case.

Sometimes when I am using faker I would like to be able to generate sentences, lines and so on with a specific length (range).

I think, it would get confusing if the `length` parameter does not refer to the length of the entire string, e.g. of lines.

Commit:
chore(lorem): simplify code and remove unnecessary tests

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

Commit:
feat(lorem): fix bug in text method

Commit:
feat(lorem): improve text method

Commit:
Merge branch 'next' into feat/lorem/issue-1546/add_character_length_parameter_to_lorem.text

ejcheng commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +321,33 @@ export class LoremModule {
+            | number
+            | {
+                /**
+                 * The minimum length of text to generate.
```
```suggestion
                 * The minimum amount of characters to generate.
```

ejcheng commented on src/modules/lorem/index.ts:
```diff
@@ -318,10 +321,33 @@ export class LoremModule {
+                 */
+                min: number;
+                /**
+                 * The maximum length of text to generate.
```
```suggestion
                 * The maximum amount of characters to generate.
```

ejcheng commented on src/modules/lorem/index.ts:
```diff
@@ -310,6 +310,9 @@ export class LoremModule {
    * Generates a random text based on a random lorem method.
    *
+   * @param options The options for the text to generate or options.length as quick primitive argument.
+   * @param options.length The length of text to generate as number or range.
```
```suggestion
   * @param options.length The length of text (in characters) to generate as number or range.
```

ST-DDT commented:
**Team Decision**

- We accept this feature request. Please rebase it and work on the suggestions we point out.

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
+      options = { length: options };
+    }
+
+    if (options.length == null) {
+      return `${this[method]()}`;
+    }
+
+    const length = this.faker.helpers.rangeToNumber(options.length);
```
Please use our standard way of decstructuring options:
```suggestion
    const { length: lengthOrRange } = options;

    if (lengthOrRange == null) {
      return `${this[method]()}`;
    }

    const length = this.faker.helpers.rangeToNumber(lengthOrRange);
```

ST-DDT commented on src/modules/lorem/index.ts:
```diff
@@ -332,7 +358,26 @@ export class LoremModule {
+      text = `${text}${this[method]()} `;
+    } while (text.length < length);
+
+    return `${text.substring(0, length).replace(/.$/, '.')}`;
```
Please just return the value without dot at the end.

```suggestion
    return text.substring(0, length);
```

ST-DDT commented on test/modules/lorem.spec.ts:
```diff
@@ -325,6 +325,53 @@ describe('lorem', () => {
+          expect(actual).toBe('');
+        });
+
+        it('should return a dot for length one', () => {
+          const actual = faker.lorem.text({ length: 1 });
+
+          expect(actual).toBeTypeOf('string');
+          expect(actual).toBe('.');
+        });
```
This test is no longer needed,

ST-DDT commented:
Closed as stale. We can reopen this when you want to continue with it.

bmulholland commented:
I just wrote a custom function for this and would love to have it baked into faker. It looks like the original submitter deleted their Github account. I'd be open to addressing the comments and submitting a fresh PR. Would that be accepted, @ST-DDT ?

xDivisionByZerox commented:
> I just wrote a custom function for this and would love to have it baked into faker. It looks like the original submitter deleted their Github account. I'd be open to addressing the comments and submitting a fresh PR. Would that be accepted, @ST-DDT ?

@bmulholland feel free to open another PR. We always love when the community gets involved. Make sure to read our contribution guide before hand and like the original issue.
